### PR TITLE
Add plots for displaced muon quantities to uGMT DQM

### PR DESCRIPTION
**Pull request by dinyar**
#### PR description:
In Run-3 displaced muon data are received from BMTF and forwarded to the uGT. This PR adds plots for these quantities to the uGMT DQM.

#### PR validation:
Ran the DQM on lxplus and verified that the new plots showed up. Checked that the old plots are still there and populated.

### CODE REVIEW

**Comment by dinyar** [comment]
attn @rekovic

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
@dinyar just for my understanding, the new plots should monitor Run3 smaples, but they seem to be added on almost any sample, even Run1 ones
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_2_X_2020-10-27-1100+7a6eb7/39605/dqm-histo-comparison-summary.html
Is this expected?

**Comment by dinyar** [comment]
Hi @jfernan2,

Yes, that's true. Is there a way to avoid them being generated for older samples? We need them for the run-3 online DQM and I'd say also relvals, but otherwise not.

Cheers,
Dinyar

**Comment by jfernan2** [comment]
You would need to create a new L1T DQM module with just the Run3 plots and then enable in DQMOffline config file like (e.g.) GEM is doing:
https://github.com/cms-sw/cmssw/blob/99948787633a713daccc6ee28b96b957ea8ed958/DQMOffline/Configuration/python/DQMOffline_cff.py#L64

**Comment by dinyar** [comment]
Hi @jfernan2,

thanks for the pointer! I would prefer if we didn't need another module as the plots should be part of the usual uGMT DQM. I have therefore put the switch based on Era into the config files for the online DQM.

One thing I'm a bit confused by is that when testing this I needed to add the lines in https://github.com/cms-sw/cmssw/pull/31951/files#diff-a4999a2c5fd36ad29c5f8c7ecba5d505fd08831bec9c4aac168f9732555d37f6R25-R28 so that the displaced muon data plots show up for Run2_2018. I would have assumed that if I don't explicitly set the behaviour for an Era the default behaviour (i.e. from https://github.com/cms-sw/cmssw/pull/31951/files#diff-a4999a2c5fd36ad29c5f8c7ecba5d505fd08831bec9c4aac168f9732555d37f6R14) would be used. Am I misunderstanding something?

Cheers,
Dinyar

**Comment by jfernan2** [comment]
Hi @dinyar 
I believe you should set the other way around: if 2018 or Run3, then:
displacedQuantities = cms.untracked.bool(True)

Usually Eras modify to set the new conditions, not the old ones

**Comment by dinyar** [comment]
Ah, alright thanks for that! Just to confirm: What would the online DQM then do? We certainly want the new plots in the online DQM  from now onwards. The Era modifier from L1T references Run3_2021, I'd like to avoid having to add Era statements as Run-3 progresses.

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
@dinyar yes, for Online DQM we need a backport to 11_1_X to be used in next MWGR, and no Era settings: we just run what is in the release, in your case:
https://github.com/cms-sw/cmssw/blob/master/DQM/Integration/python/clients/l1tstage2_dqm_sourceclient-live_cfg.py

**Comment by jfernan2** [comment]
@dinyar now there are 26 new plots in Run3 samples but there are also 4 new plots in 2018, 2017 and 2016 samples
I don't understand why:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_2_X_2020-10-29-1100+7a6eb7/39643/dqm-histo-comparison-summary.html

File: DQM/L1TMonitor/python/L1TStage2uGMT_cff.py [hunk 1]
```diff
@@ -100,6 +125,7 @@
     summaryTitle = cms.untracked.string("Summary of comparison between BMTF output muons and uGMT input muons from BMTF"),
     ignoreBin = cms.untracked.vint32(ignoreBins['Bmtf']),
     verbose = cms.untracked.bool(False),
+    isBmtf = cms.untracked.bool(True),
```
Comment (jfernan2) [inline DQM/L1TMonitor/python/L1TStage2uGMT_cff.py:128]: Might this be the responsible ?

**Comment by dinyar** [comment]
Ah right, sorry for that I missed it. Should be fine now hopefully.

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by silviodonato** [comment]
@dinyar In addition to the new plots,  there are some changes in the plots. 
Eg. 
from
image
https://tinyurl.com/yywrnnn7
to
image
https://tinyurl.com/y2ednl8n

Could you comment on this?

**Comment by dinyar** [comment]
Hi @silviodonato,

ah yes, this was a bug, apologies for that. I fixed the code and tested that workflow locally and it should be fine now. I also make changes to make sure these kind of mistakes a bit less likely in the future.

Cheers,
Dinyar

**Comment by silviodonato** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
@dinyar  can you please make a last check that all the changes inrtroduced are what you expect?
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_2_X_2020-11-02-2300+7a6eb7/39697/dqm-histo-comparison-summary.html
Thanks

**Comment by dinyar** [comment]
Hi @jfernan2,

I had a look and things look as expected now. (Some mismatches are seen in the Run-3 plots, but these can be explained by the fact that the BMTF packer hasn't been updated to work for the Run-3 data format yet.) 

One thing I noticed is that the first time I entered some of the folders I got and icon with three exclamation marks instead of a plot. When I went into the folder above and re-entered the expected plot was there. I assume this was some glitch in the machinery and not in my code, but wanted to mention it for completeness.

Cheers,
Dinyar

**Comment by jfernan2** [comment]
> One thing I noticed is that the first time I entered some of the folders I got and icon with three exclamation marks instead of a plot. When I went into the folder above and re-entered the expected plot was there. I assume this was some glitch in the machinery and not in my code, but wanted to mention it for completeness.

Yes, this is a temporary glitch from DQM GUI, new GUI in development is indeed fixing that behaviour. 

I proceed to approve the PR given your green light. Please do not forget to create a backport in 11_1_X to have these changes in the next MWGR

**Comment by jfernan2** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**